packages/margin_trading/sources/margin_manager.move

```diff
@@ -251,6 +264,134 @@ public fun repay_quote<BaseAsset, QuoteAsset>(
+public fun prove_and_destroy_request<BaseAsset, QuoteAsset, DebtAsset>(
+    margin_manager: &MarginManager<BaseAsset, QuoteAsset>,
+    registry: &MarginRegistry,
+    debt_margin_pool: &MarginPool<DebtAsset>,
+    pool: &Pool<BaseAsset, QuoteAsset>,
+    base_price_info_object: &PriceInfoObject,
+    quote_price_info_object: &PriceInfoObject,
+    clock: &Clock,
+    request: Request,
+) {
+    assert!(request.margin_manager_id == margin_manager.id(), EInvalidMarginManager);
+    assert!(margin_manager.deepbook_pool == pool.id(), EIncorrectDeepBookPool);
+
+    let risk_ratio = margin_manager
+        .manager_info<BaseAsset, QuoteAsset, DebtAsset>(
+            registry,
+            debt_margin_pool,
+            pool,
+            base_price_info_object,
+            quote_price_info_object,
+            clock,
+        )
+        .risk_ratio;
+    let pool_id = pool.id();
+    if (request.request_type == BORROW) {
+        assert!(registry.can_borrow(pool_id, risk_ratio), EBorrowRiskRatioExceeded);
+    } else if (request.request_type == WITHDRAW) {
+        assert!(registry.can_withdraw(pool_id, risk_ratio), EWithdrawRiskRatioExceeded);
+    };
+
+    let Request {
+        margin_manager_id: _,
+        request_type: _,
+    } = request;
+}
+
+/// Risk ratio = total asset in USD / (total debt and interest in USD)
+/// Risk ratio above 2.0 allows for withdrawal from balance manager, borrowing, and trading
+/// Risk ratio between 1.25 and 2.0 allows for borrowing and trading
+/// Risk ratio between 1.1 and 1.25 allows for trading only
+/// Risk ratio below 1.1 allows for liquidation
+/// These numbers can be updated by the admin. 1.25 is the default borrow risk ratio, this is equivalent to 5x leverage.
+public fun manager_info<BaseAsset, QuoteAsset, DebtAsset>(
+    margin_manager: &MarginManager<BaseAsset, QuoteAsset>,
+    registry: &MarginRegistry,
+    debt_margin_pool: &MarginPool<DebtAsset>,
+    pool: &Pool<BaseAsset, QuoteAsset>,
+    base_price_info_object: &PriceInfoObject,
+    quote_price_info_object: &PriceInfoObject,
+    clock: &Clock,
+): ManagerInfo {
+    assert!(margin_manager.deepbook_pool == pool.id(), EIncorrectDeepBookPool);
+
+    let (base_debt, quote_debt, base_asset, quote_asset) = calculate_debt_and_assets<
+        BaseAsset,
+        QuoteAsset,
+        DebtAsset,
+    >(
+        margin_manager,
+        pool,
+        debt_margin_pool,
+    ).position_info();
+
+    // Calculate debt in USD
+    let base_usd_debt = if (base_debt > 0) {
+        calculate_usd_price<BaseAsset>(
+            base_price_info_object,
+            registry,
+            base_debt,
+            clock,
+        )
+    } else {
+        0
+    };
+    let quote_usd_debt = if (quote_debt > 0) {
+        calculate_usd_price<QuoteAsset>(
+            quote_price_info_object,
+            registry,
+            quote_debt,
+            clock,
+        )
+    } else {
+        0
+    };
```

Comment:
wdyt of calculating both, and in either `calculate_debt_and_assets` or below here asserting that there is only non-zero debt in one?